iothub/device/src/IotHubBaseClient.cs

```diff
@@ -579,6 +609,13 @@ internal async Task<MessageAcknowledgement> OnMessageReceivedAsync(IncomingMessa
 
                 return MessageAcknowledgement.Abandon;
             }
+            catch (Exception ex)
+            {
+                if (Logging.IsEnabled)
+                    Logging.Error(this, $"User code threw exception: {ex}\nAbandoning message with message Id: {message.MessageId}.", nameof(OnMessageReceivedAsync));
```

Comment:
```suggestion
                    Logging.Error(this, $"Abandoning message with message Id: {message.MessageId} because user code threw exception: {ex}.", nameof(OnMessageReceivedAsync));
```